enterprise_access/apps/subsidy_access_policy/models.py

```diff
@@ -591,10 +592,27 @@ def can_redeem(self, lms_user_id, content_key, skip_customer_user_check=False):
 
         # learner not associated to enterprise
         if not skip_customer_user_check:
-            if self.lms_api_client.get_enterprise_user(self.enterprise_customer_uuid, lms_user_id) is None:
+            learner_record = self.lms_api_client.get_enterprise_user(self.enterprise_customer_uuid, lms_user_id)
+            if not learner_record:
                 self._log_redeemability(False, REASON_LEARNER_NOT_IN_ENTERPRISE, lms_user_id, content_key)
                 return (False, REASON_LEARNER_NOT_IN_ENTERPRISE, [])
 
+            associated_group_uuids = set(learner_record.get('enterprise_group', []))
+
+            policy_groups = PolicyGroupAssociation.objects.filter(subsidy_access_policy=self).all()
+            policy_groups_uuids = set(  # pylint: disable=consider-using-set-comprehension
+                [str(group.enterprise_group_uuid) for group in policy_groups]
+            )
+
+            # if there are groups associated with the policy, if not we want to be backwards compatible with
+            # policies which haven't implemented groups yet
+            if len(policy_groups) > 0:
+                uuid_overlap = associated_group_uuids.intersection(policy_groups_uuids)
+                # if any of the user's associated groups are connected to the policy
+                if len(uuid_overlap) == 0:
+                    self._log_redeemability(False, REASON_LEARNER_NOT_IN_ENTERPRISE_GROUP, lms_user_id, content_key)
+                    return (False, REASON_LEARNER_NOT_IN_ENTERPRISE_GROUP, [])
+
```

Comment:
It would be neat to extract this entire chunk of code into a method like `SubsidyAccessPolicy.includes_learner(lms_user_id)` - then we could reuse this logic down in the `credit_available()` method of this same model.  You could also simplify this a little bit by relying purely on the Django ORM to do the intersection for you
```python
def includes_learner(self, lms_user_id):
    learner_record = self.lms_api_client.get_enterprise_user(self.enterprise_customer_uuid, lms_user_id)
    if not learner_record:
        return False, REASON_LEARNER_NOT_IN_ENTERPRISE

    associated_group_uuids = set(learner_record.get('enterprise_group', []))
    # if there are no policy groups, return early
    if not PolicyGroupAssociation.objects.filter(subsidy_access_policy=self).exists():
        return True, None

    # if no association for this learner's group(s), return false
    if not PolicyGroupAssociation.objects.filter(
        subsidy_access_policy=self, 
        enterprise_group_id__in=associated_group_uuids,
    ).exists():
        return False, REASON_LEARNER_NOT_IN_ENTERPRISE_GROUP

    # otherwise, return true
    return True, None
```
then up here in `can_redeem()`:
```python
if not skip_customer_user_check:
    included_in_policy, reason = self.includes_learner(lms_user_id):
    if not included_in_policy:
        self._log_redeemability(False, reason, lms_user_id, content_key)
        return (False, reason, [])
```